[OSJC-235] DefaultClient equals even if token differs

adietish opened the pull request.

adietish commented:
@jcantrill plz review

adietish commented:
[test]

jcantrill commented on src/main/java/com/openshift/restclient/authorization/TokenAuthorizationStrategy.java:
```diff
@@ -38,5 +38,19 @@ public void accept(IAuthorizationStrategyVisitor visitor) {
 
-
+	@Override
+	public int hashCode() {
```
Why is this so complicated and not just either the hashcode of the token or String.hashCode()

adietish replied:
@jcantrill dont want to rely on the token by choice, since I consider 2 authentication strategies with different tokens to be identical

adietish commented:
updated the approach:
the DefautClient now wont compare tokens but the strategies. All strategies now report being equal upon comparison of the username (only).
This implies us to update the username in the TokenAuthorizationStrategy once we successfully connected to the server

adietish commented:
btw. I also:
- dropped the untested/unused KerbrosAuthenticationStrategy
- extracted http constants to IHttpConstants (so that we can drop the deprecated IHttpClient interface in classes that relied on it for those constants only).

adietish commented on src/main/java/com/openshift/restclient/authorization/AbstractAuthorizationStrategy.java:
```diff
@@ -0,0 +1,54 @@
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```
now strategies only rely on the username for equality. Kept username unmutable.

adietish commented on src/main/java/com/openshift/internal/restclient/DefaultClient.java:
```diff
@@ -534,22 +535,17 @@ public boolean equals(Object obj) {
+			if (other.strategy == null) {
 				return false;
+			}
+			return ObjectUtils.equals(strategy, other.strategy);
```
DefaultClient#equals wont rely on the token, only on strategy equality.
Strategies only rely on the username for equality (see comment further down).

jcantrill commented:
LGTM

jcantrill commented:
[merge]
